scripts/generate_climos

```diff
@@ -18,7 +18,8 @@ def main(args):
             except Exception as e:
                 logger.info('{}: {}'.format(e.__class__.__name__, e))
             else:
-                logger.info('climo_periods: {}'.format(input_file.climo_periods.keys()))
+                periods = input_file.climo_periods.keys() & args.climo
```

Comment:
`args.climo` is a list if the user specifies it (but not if not; you supply the default value of `standard_climo_periods().keys()`, which is of type `dict_keys`. The operator `&` isn't defined for operands of type `list`, though it does work as expected for `dict_keys`.

Reply:
I'm pretty sure that it works as long as one argument is a `dict_keys`, no?

```
$ python3
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> {1: 1, 2: 2, 3: 3}.keys() & [1, 2]
{1, 2}
>>> [1, 2] & {1: 1, 2: 2, 3: 3}.keys()
{1, 2}
>>> [1, 2] & {1: 1, 2: 2, 3: 3}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for &: 'list' and 'dict'
```

Reply:
Okay, that's weird. I got an error the first time I tried this, but now it shows you are right. Not a good day for me.

Reply:
Ah, I forgot the function invocation parentheses. Definitely doesn't work with a signature of `function` & `list` :) :(